services/storage/src/simcore_service_storage/s3_client.py

```diff
@@ -237,17 +242,22 @@ async def complete_multipart_upload(
         file_id: SimcoreS3FileID,
         upload_id: UploadID,
         uploaded_parts: list[UploadedPart],
+        sha256_checksum: SHA256Str | None,
     ) -> ETag:
-        response = await self.client.complete_multipart_upload(
-            Bucket=bucket,
-            Key=file_id,
-            UploadId=upload_id,
-            MultipartUpload={
+        inputs: dict[str, Any] = {
+            "Bucket": bucket,
+            "Key": file_id,
+            "UploadId": upload_id,
+            "MultipartUpload": {
                 "Parts": [
                     {"ETag": part.e_tag, "PartNumber": part.number}
                     for part in uploaded_parts
                 ]
             },
+            "ChecksumSHA256": None if sha256_checksum is None else str(sha256_checksum),
```

Comment:
so if I get that right, you still replace the ETag from S3 using this right?
what happens if we copy the data in any way? this will be lost or not?
vs setting it via Metadata field?

Reply:
If I understand the [documentation](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3/client/complete_multipart_upload.html) correctly, this should not replace the etag. This is just a way to add a manually (meaning, on the client side) computed checksum to the upload as metadata. I can try to experiment a bit with it, to see if it behaves as wished